File: src/test/scala/com/exasol/cloudetl/source/SourceTest.scala
package com.exasol.cloudetl.source

class SourceTest extends AbstractSourceTest {

  override val format: String = "avro"

  test("apply throws if file format is not supported") {
    val path = new org.apache.hadoop.fs.Path(s"$resourceDir/sales10.avro")
    val thrown = intercept[IllegalArgumentException] {
      getSource(path, "file")
    }
    assert(thrown.getMessage().startsWith("E-CSE-21"))
    assert(thrown.getMessage().contains("Storage format 'FILE' is not supported."))
  }

}

File: src/test/java/com/exasol/cloudetl/source/SourceTest.java
package com.exasol.cloudetl.source;

import static org.junit.jupiter.api.Assertions.*;

import org.apache.hadoop.fs.Path;
import org.junit.jupiter.api.Test;

class SourceTest extends AbstractSourceTest {
    @Override
    protected String getFormat() {
        return "avro";
    }

    @Test
    void createThrowsIfFileFormatIsNotSupported() {
        final Path path = new Path(this.resourceDir + "/sales10.avro");
        final IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class, () -> getSource(path, "file"));
        assertTrue(thrown.getMessage().startsWith("E-CSE-21"));
        assertTrue(thrown.getMessage().contains("Storage format 'FILE' is not supported."));
    }
}
